src/middle/Validation.ml

```diff
@@ -0,0 +1,55 @@
+include Validation_intf
```

Comment:
Actually, could you add a little bit of documentation to the Validation monad? A minimum would be a comment at the top to explain what it's supposed to do (e.g. accumulate errors).

Reply:
Will do - good shout.